### PR TITLE
feature(customization): how you integrate with salt-desktop

### LINKED ISSUE #14: Easy integration into your site
While designed for easy reuse and extension, a reference solution is undocumented. Therefore It may be unclear how to integrate with your sites.  Here is a simple three-step guide explaining how a site named BigCorp could integrate salt-desktop....... a generic solution.

** 1. Your site-specific repo: https://git.bigcorp.com/bigcorp-desktop.git**
```
bigcorp-desktop/
      profiles/
          bigcorp.sls
          bigcorp/
              profleA
              profileB
      sbin/
      bigcorp.sh
      README.md
```
- profiles/ directory contains site-specific configuration customizations.
- sbin directory contains site-specific `devsetup.sh` and `menu.py`
- README.md contains site-specific usage howto....

** 2.  Your site specific readme: https://git.bigcorp.com/bigcorp-desktop/raw/README.md **

    ====== readme start ======

  Salt Desktop orchestrates useful software onto Linux/MacOS without fuss. 

  _Setup_

  `curl -o salt.sh https://raw.githubusercontent.com/saltstack-formulas/salt-desktop/master/bin/salt.sh && sudo bash salt.sh`
  `sudo -s`
  `rm -fr /srv/salt && git clone https://git.bigcorp.com/bigcorp-desktop.git /srv/salt`
  `cd /srv/salt`
  `./bigcorp.sh`
  `exit`

For generic usage, see https://github.com/saltstack-formulas/salt-desktop#stack-profiles

_BigCorp next step_

bla bla bla  bla bla

    ==== readme end =====

** 3. Your site-specific `bigcorp.sh` script**
```
#!/usr/bin/env bash
git config user.email "not@important.com"
git config user.name "not important"
git init
git remote add salt https://github.com/saltstack-formulas/salt-desktop.git
git pull salt master --allow-unrelated-histories -f >/dev/null
if (( $? == 0 ))
then
    mv ./sbin/* ./bin/
    mv ./profiles/bigcorp.sls ./profiles/config.sls
    git init
else
    [[ -x "/usr/bin/xcode-select" ]] && /usr/bin/xcode-select --install
    echo "Error - something is not right"
    echo "Ensure your OS is uptodate (git 2.9 or later)"
    exit 1 
fi
```

Maybe formula should be expanded to add this basic **_integration_**  guidance.

### PR DESCRIPTION
**Pull request by noelmcloughlin**
This PR resolves #14 by providing a framework for integrating/personalization.

1. Create and clone your personalized  *salt* repo with at least the following structure.

      **/scripts/overlay-salt.sh**         => sample provided in ./contrib/ directory
      **/profiles/**                                => your custom SLS (high) states to be copied to file_roots
      **/configs/**                                 => your custom SLS pillar data to be copied to pillar_roots

2. Run the sample script to merge your configuration into salt-desktop.

       `cd <repo-name> && sudo ./scripts/overlay-salt.sh -i salt`

Now refer to README.   
Basically you run `sudo /usr/local/bin/salter.sh -i <profile-name>` to install stuff.

### CODE REVIEW

**Comment by noelmcloughlin** [comment]
Selfie merge - I've been testing this all day on MacOS.
